### ISSUE TITLE
If launcher, desktop wallpaper offset at startup

### ISSUE THREAD

**Issue by dmcdivitt** [farmerbb/Taskbar#46]
I have android 7.1.2 on several TX92 TV PCs. If Taskbar is the default launcher and loaded at startup, wallpaper will occupy the right two-thirds of the screen, only. If Nova Launcher is opened, then closed, wallpaper will properly align with the entire screen.

When using Nova Launcher as the default launcher and loaded at startup, wallpaper corrects as soon as Nova Launcher loads.

This may indicate maintenance needed by Taskbar when it loads as default launcher. Other launchers were tried and they also correct the wallpaper position.

**Comment by dmcdivitt** [comment]
I have the program Cool Daily WP that fetches the Bing wallpaper each day. It must either crop or resize the image when retrieved. It doesn't do that correctly and saves a 1920x1080 image with borders. The image saved with borders is the same size as the original. Android shows as much as it can of the image, a good part of which is the border. When Nova Launcher loads it centers and sizes the image to fit. So the problem is with Cool Daily WP.

No need to address this with Taskbar, I guess.

**Comment by dmcdivitt** [comment]
I closed this issue, to try other wallpaper apps, thinking Cool Daily WP is the problem. It seems other wallpaper apps don't work as well. Reading about this, the android wallpaper must be saved at twice the width. The android OS distributes this out to three possible panes. Cool Daily WP does what is needed by adding a margin to each side of the image so it will display properly.

So I reopened the issue. When Taskbar is the launcher, it is not displaying the correct wallpaper pane, so it is a problem with Taskbar. If Taskbar is just fetching the wallpaper image and displaying it, that is wrong.

**Comment by farmerbb** [comment]
I haven't come across this issue on any device I've tested Taskbar on (sounds like it could be an issue with the Android build your TV stick runs), but, I can take a look at what other launchers might be doing and see what I can do to improve wallpaper handling.  (It looks like there are some [`WallpaperManager`](https://developer.android.com/reference/android/app/WallpaperManager.html) APIs that might help with this scenario)

**Comment by dmcdivitt** [comment]
From what I've read there are three wallpaper panes: a half-width left 
pane, a full size center pane, and a half-width right pane. One image 
facilitates all three panes.

There is a current pane, or the pane currently being displayed. If the 
current pane is the left or right pane, half of the center pane is used 
in addition, because left and right are only half-width.

If I open Nova Launcher then close it (stop the process), Nova Launcher 
makes the center pane the current pane. From that time on, any wallpaper 
change I make with Cool Daily WP will work correctly. This is because 
the center pane is still the current pane. Cool Daily WP pads the 
wallpaper image on both sides so wallpaper will display properly as the 
center pane.

If I reboot, the left pane becomes the current pane. Because the 
wallpaper image has been padded, the left half of the screen is blank, 
the wallpaper appears only on the right half, and the wallpaper appears 
shifted.

Opening and closing Nova Launcher selects the center pane as the current 
pane.

For the Taskbar program, what might be done is provide a wallpaper pane 
selection option: left, center, or right. Taskbar would then enforce 
that selection when it is the launcher and home is selected.

I did some reading. The functionality I described is standard for all 
applications and the wallpaper image described is standard.

--
yes, I dare to be subjective!

**Comment by farmerbb** [comment]
Ok, I just committed a fix that should center the wallpaper offset when Taskbar is set as the launcher.  I'll release this soon as part of version 3.8.1.

**Comment by dmcdivitt** [comment]
In the play store it shows v 3.8.1 for the pro version, which I have. There is no help/about in the app showing version info. You may want to add that. I do not see any difference in wallpaper. Still offset when booted and sometimes when changing wallpaper.

Sure am happy to have this program. Not bitching at all!

**Comment by farmerbb** [comment]
Glad you're enjoying it!

I just released another update today that lets Taskbar specify to the system the correct size wallpaper it needs when wallpaper is changed (something I had forgot to include earlier).  Taskbar should now be doing everything it can to ensure that wallpaper looks correctly when it is set as the default launcher.

If you're still having issues, try this:
* Ensure that no other launchers are running besides Taskbar (they may try to override the wallpaper positioning)
* Toggle off and on the "Replace home screen" checkbox under "Advanced settings", which will restart the activity that controls the wallpaper

**Comment by dmcdivitt** [comment]
Yes, that did it. As soon as I updated with google play the wallpaper 
corrected. Thanks!

--
yes, I dare to be subjective!
